hardware_interface/src/resource_manager.cpp

```diff
@@ -1283,6 +1304,58 @@ HardwareReadWriteStatus ResourceManager::read(
   return read_write_status;
 }
 
+void ResourceManager::publish_all_interface_values() const
+{
+  control_msgs::msg::DynamicInterfaceValues interface_values;
+  interface_values.header.stamp = resource_storage_->clock_interface_->get_clock()->now();
+
+  control_msgs::msg::InterfaceValue state_interface_values;
+  for (const auto & state_interface_name : resource_storage_->available_state_interfaces_)
+  {
+    try
+    {
+      const auto state_interface_value =
+        resource_storage_->state_interface_map_.at(state_interface_name).get_value();
+      state_interface_values.interface_names.push_back(state_interface_name);
+      state_interface_values.values.push_back(state_interface_value);
+    }
+    catch (const std::out_of_range & e)
+    {
+      RCUTILS_LOG_WARN_NAMED(
+        "resource_manager",
+        "State interface '%s' is in available list, but could not get the interface "
+        "state_interface_map_ (std::out_of_range exception thrown).",
+        state_interface_name.c_str());
+    }
+  }
+
+  control_msgs::msg::InterfaceValue command_interface_values;
+  for (const auto & command_interface_name : resource_storage_->available_command_interfaces_)
+  {
+    try
+    {
+      const auto command_interface_value =
+        resource_storage_->command_interface_map_.at(command_interface_name).get_value();
+      command_interface_values.interface_names.push_back(command_interface_name);
+      command_interface_values.values.push_back(command_interface_value);
+    }
+    catch (const std::out_of_range & e)
```

Comment:
`e` is unused here, so:
```suggestion
    catch (const std::out_of_range &)
```

Reply:
use the proposal from below